packages/blade/src/components/Amount/Amount.tsx

```diff
@@ -0,0 +1,271 @@
+const getFlooredFixed = (value: number, precision: number): number => {
+  const factor = 10 ** precision;
+  const roundedValue = Math.floor(value * factor) / factor;
+  return Number(roundedValue.toFixed(precision));
+};
+
+const addCommas = (amountValue: number, currency: Currency): string => {
+  const locale = currencyLocaleMapping[currency];
+  return amountValue.toLocaleString(locale);
+};
+
+const getHumanizedAmount = (amountValue: number, currency: Currency): string => {
+  const abbreviations = currencyAbbreviationsMapping[currency];
+
+  const abbreviation = abbreviations.find((abbr) => amountValue >= abbr.value);
+  if (abbreviation) {
+    amountValue = amountValue / abbreviation.value;
+    const formattedAmountValue = getFlooredFixed(amountValue, 2);
+    return addCommas(formattedAmountValue, currency) + abbreviation.symbol;
+  } else {
+    return amountValue.toString();
+  }
+};
+
+type FormatAmountWithSuffixType = {
+  suffix: AmountProps['suffix'];
+  value: number;
+  currency: Currency;
+};
+
+const formatAmountWithSuffix = ({
+  suffix,
+  value,
+  currency,
+}: FormatAmountWithSuffixType): string => {
+  switch (suffix) {
+    case 'decimals': {
+      const decimalNumber = getFlooredFixed(value, 2);
+      return addCommas(decimalNumber, currency);
+    }
+    case 'humanize': {
+      return getHumanizedAmount(value, currency);
+    }
+    case 'none': {
+      return addCommas(getFlooredFixed(value, 0), currency);
+    }
+    default:
+      return addCommas(getFlooredFixed(value, 0), currency);
+  }
+};
+
+const getCurrencyWeight = (
+  isAffixSubtle: NonNullable<AmountProps['isAffixSubtle']>,
+  size: NonNullable<AmountProps['size']>,
+): 'bold' | 'regular' => {
+  if (isAffixSubtle || size.startsWith('bold')) return 'bold';
+  return 'regular';
+};
+
+const Amount = ({
+  value,
+  suffix = 'decimals',
+  size = 'heading-small',
+  isAffixSubtle = true,
+  intent,
+  prefix = 'currency-symbol',
+  testID,
+}: AmountProps): ReactElement => {
+  if (isNaN(value)) {
+    throw new Error('[Blade: Amount]: `value` prop must be of type `number` for Amount.');
+  }
+
+  // This will be added to prop and can be switched to a currency
+  const currency = 'INR';
+  const currencyPrefix = currencyPrefixMapping[currency][prefix];
+  const renderedValue = formatAmountWithSuffix({ suffix, value, currency });
+  const { amountValueColor, affixColor } = getTextColorProps({
+    intent,
+  });
+
+  const currencyColor = isAffixSubtle ? affixColor : amountValueColor;
+  const currencyFontSize = getAffixFontSize(isAffixSubtle, size);
+  const currencyWeight = getCurrencyWeight(isAffixSubtle, size);
+
+  return (
+    <BaseBox
+      paddingLeft="spacing.2"
+      paddingRight="spacing.2"
+      {...metaAttribute({ name: MetaConstants.Amount, testID })}
+    >
+      <BaseBox
+        display="flex"
+        alignItems="baseline"
+        paddingLeft="spacing.2"
+        paddingRight="spacing.2"
```

Comment:
Why is this padding here? that's a padding of 4px on each side, which is not on figma either.